lib/cdo/lesson_import_helper.rb

```diff
@@ -19,34 +19,47 @@ module LessonImportHelper
   # }
   # @param [Lesson] lesson - Code Studio Lesson object to update.
   # @param [Hash] cb_lesson_data - Lesson and activity data to import.
-  def self.update_lesson(lesson, cb_lesson_data = {})
+  def self.update_lesson(lesson, models_to_import = [], cb_lesson_data = {})
     raise unless [:development, :adhoc, :levelbuilder].include? rack_env
     raise unless lesson.script.hidden
 
+    raise "Must specify models to import" if models_to_import.blank?
+
     # course version id should always be present for CSF/CSD/CSP 2020 courses.
     course_version_id = lesson.script&.get_course_version&.id
     raise "Script must have course version" unless course_version_id
 
     lesson_levels = lesson.script_levels.reject {|l| l.levels[0].type == 'CurriculumReference'}
-    if cb_lesson_data.empty?
+    if cb_lesson_data.empty? && models_to_import.include?('Lesson')
       lesson.lesson_activities = update_lockable_lesson(lesson_levels, lesson.id)
       lesson.script_levels = []
     else
-      lesson.name = cb_lesson_data['title']
-      lesson.overview = cb_lesson_data['teacher_desc']
-      lesson.student_overview = cb_lesson_data['student_desc']
-      lesson.purpose = cb_lesson_data['cs_content']
-      lesson.preparation = cb_lesson_data['prep']
-      lesson.creative_commons_license = cb_lesson_data['creative_commons_license']
-      lesson.assessment_opportunities = cb_lesson_data['assessment'] unless cb_lesson_data['assessment'].blank?
-      lesson.objectives = cb_lesson_data['objectives'].map do |o|
-        Objective.new(key: SecureRandom.uuid, description: o["name"])
+      if models_to_import.include?('Lesson')
```

Comment:
nit:  this is just my opinion, but I think it would be a little easier to read if you did the `if models_to_import.include?('Lesson')` as the outer conditional, and `if cb_lesson_data.empty?` as the inner one.

Reply:
That makes a lot of sense! I'll rearrange

Reply:
Oh actually, this doesn't work. Lockable lessons cause other blocks to crash because the entry in the hash doesn't exist. We shouldn't be trying to import _anything_ other than lesson information for lockable lessons so I'm going to switch this back (mostly)

Reply:
ohh, I see. I don't know if I have just become totally allergic to nested ifs, but if you are on board with it another option would be to do an early return from this method `if cb_lesson_data.empty?`. otherwise how you have it is fine. "as simple as possible, but not simpler". :-)

Reply:
Haha I went back and forth between an early return and the way it is now. They both feel kind of bad to me so I'm not particularly attached to either

Reply:
as the PR author, you have the burden of deciding :-)

Reply:
I'm going to leave as-is then. It's very slightly clearer this way to me